### DISCUSSION TITLE
Manually adding an object to the Physics world without using hooks

### DISCUSSION THREAD

**Discussion by shampliu** [pmndrs/use-cannon#92] [General] [answered]
I have the same use case as [this question](https://github.com/react-spring/use-cannon/discussions/91), basically an imported .glb model with a lot of meshes. 

However, I'm wondering if there's a way to add these objects directly to the World instantiated in the `worker.js` file? This is what I came up with so far:

```
import { context } from "use-cannon";

export default function Model(props) {
  const ctx = useContext(context);
  const { nodes, materials } = useLoader(
    GLTFLoader,
    "/models/world_flat-2.glb"
  );

  const cannonProps = useMemo(() => {
    return Object.entries(nodes).map(([key, node]) => {
      if (!node || !node.geometry) {
        return null;
      }

      node.geometry.computeBoundingBox();
      const { position, rotation = [0, 0, 0], uuid, geometry } = node;
      const { min, max } = geometry.boundingBox;
      const scale = [ max.x - min.x, max.y - min.y, max.z - min.z ];

      return {
        position, rotation, uuid, scale, type: 'Dynamic'
      }
    }).filter(Boolean);
  }, [nodes]);

  useEffect(() => {
    ctx.worker.postMessage({ op: 'addBodies', type: 'Box', props: cannonProps, uuid: cannonProps.map(p => p.uuid)  })
  }, [cannonProps]);

  ...
}
```

Basically just trying to make a box object in the Cannon World based off the bounding boxes of each object in the scene. With this code, I run into an error: 

`Failed to execute 'postMessage' on 'Worker': function onRotationChange() {
quaternion.setFromEuler(rotation, false);
} could not be cloned.`

**Comment by drcmda** [comment, marked as answer]
why though? better make a component, you dont need a visual representation, take a look at the borders component for instance: https://codesandbox.io/s/ball-pit-1-mvkqs?file=/src/App.js:714-1374

**Reply by shampliu** [reply]
@drcmda is there a more programmatic way to attach those refs from `useBox` / `useSphere` that's faster than manually passing it in to all the objects in the gltjsx? 

My scene has a ton of objects that need to have physics and we might be changing up the imported `.glb` file alot.

**Reply by drcmda** [reply]
well you can of course use cannon-es. what youre asking for imperative access, but use-cannon has explicitly been written for declarative use.

**Reply by shampliu** [reply]
I see, my question now is if there's an easy way to access the Physics world created in the worker file? My use case is pretty trivial as I just need to add a bunch of objects into the world which won't move after they are placed. Is that possible to do with cannon-es while also using use-cannon for other objects?

**Reply by drcmda** [reply]
if they're the same object instancing would be the most efficient and performant, see: https://codesandbox.io/s/ball-pit-1-mvkqs though more then 200 is quite the burden because it has to figure out all the collisions. otherwise just create components for each obj.
